core/state/state_object.go

```diff
@@ -116,10 +116,16 @@ func newObject(db *StateDB, address common.Address, data types.StateAccount) *st
 		data.Root = db.GetEmptyRoot()
 		// [Scroll: END]
 	}
+	var addrHash common.Hash
+	if db.IsZktrie() {
+		addrHash = common.BytesToHash(trie.NewZktHashFromBytesOrPanic(address[:]).Bytes())
```

Comment:
I think you need to leave comment something like "we don't hash address here, because hash is done inside ZkTrie`.